File: src/main/java/com/dambra/paul/moduluschecker/valacdosFile/SortCodeRange.java
package com.dambra.paul.moduluschecker.valacdosFile;

import com.google.common.base.CharMatcher;
import com.google.common.base.Splitter;

import java.util.List;
import java.util.Optional;

public final class SortCodeRange {
    private static final Splitter splitter = Splitter.on(CharMatcher.whitespace())
            .omitEmptyStrings();

    private final int start;
    private final int end;

    public SortCodeRange(String lowest, String highest) {
        start = asInteger(lowest);
        end = asInteger(highest);
    }

    private int asInteger(String s) {
        return Integer.parseInt(s.replaceAll("^0+", ""));
    }

    static Optional<SortCodeRange> parse(String input) {
        List<String> parts = splitter.splitToList(input);

        if (parts.isEmpty()) {
            return Optional.empty();
        }

        return Optional.of(
                new SortCodeRange(
                        parts.get(WeightRow.LOWEST_SORT_CODE_INDEX),
                        parts.get(WeightRow.HIGHEST_SORT_CODE_INDEX)));
    }

    public boolean contains(String sortCode) {
        int i = asInteger(sortCode);
        return start <= i && i <= end;
    }
}

File: src/main/java/com/github/pauldambra/moduluschecker/valacdosFile/SortCodeRange.kt
package com.github.pauldambra.moduluschecker.valacdosFile

data class SortCodeRange(val lowest: String, val highest: String) {

    private val start: Int
    private val end: Int

    init {
        start = asInteger(lowest)
        end = asInteger(highest)
    }

    private fun asInteger(s: String) = Integer.parseInt(s.replace("^0+".toRegex(), ""))

    operator fun contains(sortCode: String) = asInteger(sortCode) in start..end

    companion object {

        private const val LOWEST_SORT_CODE_INDEX = 0
        private const val HIGHEST_SORT_CODE_INDEX = 1

        internal fun parse(input: String): SortCodeRange {
            val parts = input
              .split(" ")
              .filter { it.isNotBlank() }

            return SortCodeRange(
                  parts[LOWEST_SORT_CODE_INDEX],
                  parts[HIGHEST_SORT_CODE_INDEX])
        }
    }
}
